iothub/device/src/Pipeline/TransportHandlerFactory.cs

```diff
@@ -12,49 +12,43 @@ internal class TransportHandlerFactory : ITransportHandlerFactory
     {
         public IDelegatingHandler Create(PipelineContext context)
         {
-            // ProtocolRoutingDelegatingHandler configures the TransportSettings configuration
-            // which is different from TransportSettings[] element.
-            IotHubClientTransportSettings transportSettings = context.TransportSettings;
             IotHubConnectionInfo connInfo = context.IotHubConnectionInfo;
             InternalClient.OnMethodCalledDelegate onMethodCallback = context.MethodCallback;
             Action<TwinCollection> onDesiredStatePatchReceived = context.DesiredPropertyUpdateCallback;
             InternalClient.OnModuleEventMessageReceivedDelegate onModuleEventReceivedCallback = context.ModuleEventCallback;
             InternalClient.OnDeviceMessageReceivedDelegate onDeviceMessageReceivedCallback = context.DeviceEventCallback;
 
-            if (transportSettings is IotHubClientAmqpSettings)
+            if (connInfo.ClientOptions.TransportSettings is IotHubClientAmqpSettings iotHubClientAmqpSettings)
             {
                 return new AmqpTransportHandler(
                     context,
-                    connInfo,
-                    transportSettings as IotHubClientAmqpSettings,
+                    iotHubClientAmqpSettings,
                     new Func<MethodRequestInternal, Task>(onMethodCallback),
```

Comment:
PipelineContext holds the callback definitions as well, so they can also be removed from this method signature. However, the delegate -> function transformation needs to be done first, so I'll do that in the next PR.